postgres/client.js

```diff
@@ -272,12 +272,40 @@ function getMeta(key) {
  * @param {Object} value [description]
  * @return {Promise} [description]
  */
+/* eslint complexity: 0 */
 function putMeta(key, value) {
   const { schema, table } = findSchemaAndTable(key),
-    map = columnToValueMap('id', key);
+    map = columnToValueMap('id', key),
+    parsedValue = parseOrNot(value);
 
   // add meta column to map
-  columnToValueMap('meta', parseOrNot(value), map);
+  columnToValueMap('meta', parsedValue, map);
+
+  if (isPage(key)) {
+    // If value has first publish time, add it to the map
+    if (parsedValue.firstPublishTime)
+      columnToValueMap('published_at', parsedValue.firstPublishTime, map);
+
+    // If value has publish time, add it as republished date
+    if (parsedValue.publishTime)
+      columnToValueMap('republished_at', parsedValue.publishTime, map);
+
+    // If we have history data, then find the unpublish and archive events
+    if (parsedValue.history && parsedValue.history.length) {
+      const latestUnpublish = parsedValue.history
```

Comment:
You could abstract this logic into a small function
```suggestion
      const latestUnpublish = getLatestAction(parsedValue.history, 'unpublish'),
        latestArchived = getLatestAction(parsedValue.history, 'archive')
```

```js
function getLatestAction(history = [], action) {
  return history
    .filter(event => event.action === action)
    .pop() || {};
}
```